src/test/java/de/tum/cit/aet/artemis/communication/notification/NotificationScheduleServiceTest.java

```diff
@@ -91,7 +91,7 @@ void shouldCreateNotificationAndEmailAtReleaseDate() {
         exerciseRepository.saveAndFlush(exercise);
 
         instanceMessageReceiveService.processScheduleExerciseReleasedNotification(exercise.getId());
-        await().until(() -> notificationRepository.count() > sizeBefore);
+        await().until(() -> notificationTestRepository.count() > sizeBefore);
```

Comment:
I am fully aware that this is totally out of scope for this PR but it appears there's a reason why some tests regarding notifications are often times flaky. Also the `delete()` in `NotificationResourceIntegrationTest` might play a role.

Reply:
I agree, there is probably some sub-optimal use of deletes for theses tests. This should be investigated in a follow up :+1: